src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -160,6 +161,23 @@ public static void writeRandomDecimals(int size, IonType type, String path, Stri
         WriteRandomIonValues.printInfo(path);
     }
 
+    /**
+     * Write random Ion decimals into the file, and all generated data is conformed with the constraints provided by Ion schema file.
+     * @param size specifies the size in bytes of the generated file.
+     * @param type determines which type of ion data will be generated.
+     * @param path the destination of the generated file.
+     * @param format the format of output file (ion_binary | ion_text).
+     * @param constraintStruct is an IonStruct which contains the top-level constraints in Ion Schema.
+     * @throws Exception if an error occurs when building up the writer.
+     **/
+    public static void writeRandomDecimalsFromSchema(int size, IonType type, String path, String format, IonStruct constraintStruct) throws Exception {
+        File file = new File(path);
+        try (IonWriter writer = WriteRandomIonValues.formatWriter(format, file)) {
+            WriteRandomIonValues.writeRequestedSizeFile(size, writer, file, type, NO_CODE_POINT_LENGTH, NO_POINT_RANGE, NO_EXPONENT_VALUE_RANGE, NO_COEFFICIENT_DIGIT_RANGE, NO_TIMESTAMP_TEMPLATE, constraintStruct);
```

Comment:
Seeing a method with this many parameters makes me think a refactoring is probably a good idea. We talked offline about deprecating the individual options for specifying constraints manually, in favor of doing all generation via schema. I think that could help with this.

Reply:
> Seeing a method with this many parameters makes me think a refactoring is probably a good idea. We talked offline about deprecating the individual options for specifying constraints manually, in favor of doing all generation via schema. I think that could help with this.

I'll create an issue on this. The next PR will include the refactoring.